### PR TITLE
[FEATURE] Omnichannel - Contacts endpoints.

### PR DESCRIPTION
**Pull request by rafaelblink**
## Proposed changes

Creating the "contacts" end points in order to Omnichannel directory pages.

api/v1/contact ( Authentication required )
POST - Create or Update data from a contact.
GET - Get data from a specific contact.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Hotfix (a major bugfix that has to be merged asap)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/livechat/server/api/v1/contact.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { Match, check } from 'meteor/check';
+
+import { API } from '../../../../api/server';
+import { Contacts } from '../../lib/Contacts';
+
+
+API.v1.addRoute('contact', { authRequired: true }, {
+	post() {
```
Comment (renatobecker) [inline app/livechat/server/api/v1/contact.js:8]: `POST` method is used to create new objects, which means the `ID` field is not expected here. If you want to allow updating objects, you need to use `PUT`.

File: app/livechat/server/api/v1/contact.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+				name: String,
+				email: Match.Maybe(String),
+				phone: Match.Maybe(String),
+			});
+
+			const contactParams = this.bodyParams;
+			if (this.bodyParams.phone) {
+				contactParams.phone = { number: this.bodyParams.phone };
+			}
+			const contact = Contacts.saveContact(contactParams);
```
Comment (renatobecker) [inline app/livechat/server/api/v1/contact.js:21]: Why not using `Livechat.registerGuest` method?

File: app/livechat/server/api/v1/contact.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+		} catch (e) {
+			return API.v1.failure(e);
+		}
+	},
+	get() {
+		check(this.queryParams, {
+			contactId: String,
+		});
+
+		const contact = Promise.await(Contacts.findContactInfo(this.queryParams.contactId));
```
Comment (renatobecker) [inline app/livechat/server/api/v1/contact.js:33]: Why not using `LivechatVisitors.findOneById` ?

File: app/livechat/server/lib/Contacts.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
+import _ from 'underscore';
+
+import {
+	LivechatVisitors,
+} from '../../../models';
+
+
+export const Contacts = {
+	saveContact({ _id, name, email, phone } = {}) {
```
Comment (renatobecker) [inline app/livechat/server/lib/Contacts.js:9]: Why not using `Livechat.registerGuest` ?

File: app/livechat/server/lib/Contacts.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+		if (_.isEmpty(updateUser.$unset)) { delete updateUser.$unset; }
+
+		LivechatVisitors.updateById(contactId, updateUser);
+
+		const contactUpdated = LivechatVisitors.findOneById(contactId);
+
+		return contactUpdated;
+	},
+	findContactInfo(contactId) {
```
Comment (renatobecker) [inline app/livechat/server/lib/Contacts.js:57]: Why do you need to create a new method that just calls `LivechatVisitors.findOneById`? 
You could call `LivechatVisitors.findOneById`, no?

**Review by renatobecker — CHANGES_REQUESTED** [review]

File: app/livechat/server/api/v1/contact.js (hunk 2 above)
Comment (rafaelblink) [reply app/livechat/server/api/v1/contact.js:21]: registerGuest is based on token, we don't use token here.
I've created the "Contacts" service level, so he is using `LivechatVisitors.insert` method.

File: app/livechat/server/api/v1/contact.js (hunk 3 above)
Comment (rafaelblink) [reply app/livechat/server/api/v1/contact.js:33]: `Contacts` service level is using `LivechatVisitors.findOneById`

File: app/livechat/server/lib/Contacts.js (hunk 4 above)
Comment (rafaelblink) [reply app/livechat/server/lib/Contacts.js:9]: registerGuest is based on token, we don't use token here.
I've created the "Contacts" service level, so he is using `LivechatVisitors.insert` method.

File: app/livechat/server/api/v1/contact.js (hunk 2 above)
Comment (renatobecker) [reply app/livechat/server/api/v1/contact.js:23]: You can generate a `new token` and pass it as a parameter.

**Pull request closed without merging**